src/v/rpc/test/rpc_gen_cycling_test.cc

```diff
@@ -525,3 +525,253 @@ FIXTURE_TEST(version_not_supported, rpc_integration_fixture) {
 
     ss::when_all_succeed(requests.begin(), requests.end()).get();
 }
+
+/*
+ * new client, new server
```

Comment:
Elsewhere you mention RPCs being processed out of order, and the bumped transport version not necessarily reflecting the client's expected version. Is that something we can test? Would we expect the RPCs to still succeed?

Reply:
it would be difficult to test, but we might be able to. it sounds much more complicated than it really is. all it really means is that when a client dispatches a request it should use a consistent version for the life time of the request/response.